### PR TITLE
Add watchosX64 target

### PR DESCRIPTION
**Pull request by BoD**
This is a followup to https://github.com/square/okio/pull/1027 and https://github.com/square/okio/pull/1028, and adds `watchosX64` (Apple watchOS 64-bit simulator for watchOS 7.0 and newer, on x86_64 platforms)

### CODE REVIEW

**Review by martinbonnin — APPROVED** [review]

**Review by oldergod — APPROVED** [review]
Thank you

File: build-support/src/main/kotlin/platforms.kt [hunk 1]
```diff
@@ -46,7 +48,7 @@ val linuxTargets = listOf(
 
 val nativeTargets = appleTargets + linuxTargets + mingwTargets
 
-/** Note that size_t is 32-bit on all watchOS versions (ie. pointers are always 32-bit). */
+/** Note that size_t is 32-bit on legacy watchOS versions (ie. pointers are always 32-bit). */
```
Comment (swankjesse) [inline build-support/src/main/kotlin/platforms.kt:51]: Why the change to the word legacy? Arm64 is the architecture of the current Apple watch.

Comment (BoD) [reply build-support/src/main/kotlin/platforms.kt:51]: Woops sorry, you're right, I think I got confused because `watchosX64` is 64 bits and specific to recent WatchOS, but that's only the simulator, not the actual watch.
